jme3-terrain/src/main/java/com/jme3/terrain/executor/TerrainExecutorService.java

```diff
@@ -0,0 +1,108 @@
+package com.jme3.terrain.executor;
+
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * The class to provide single executor service to run background tasks of terrain staff.
+ *
+ * @author JavaSaBr
+ */
+public class TerrainExecutorService {
```

Comment:
It's just lazy singleton of global executor service for terrain usage with the option to override implementation.